core/runtime/impl/tagged_transaction_queue_impl.cpp

```diff
@@ -36,7 +36,7 @@ namespace kagome::runtime {
         executor_.call(state_code_,
                        "TaggedTransactionQueue_validate_transaction", ll));
 
-    uint32_t res_addr = res.geti64() >> 32;
+    uint32_t res_addr = res.geti64() >> 32ul;
 
     WasmMemoryStream s(memory_);
     OUTCOME_TRY(s.advance(res_addr));
```

Comment:
advance() takes uint32_t, so implicit conversion occurs, which is not good thing. 
Probably, we need static_cast<uint64_t>(res_addr);

Reply:
You mean advance takes uint64_t and I pass uint32_t? But non-narrowing conversion of and unsigned type should be okay as it is?

Reply:
I misunderstood. It's ok. 
Anyway, there is a place for doubts: 
Core API takes [minor uint32_t of returned int64_t](https://github.com/soramitsu/kagome/blob/f0eb729bd5f812cb3eec544c6bf980e9b368bf80/core/runtime/impl/core_impl.cpp#L31)

 the same time TTQ takes [major part](https://github.com/soramitsu/kagome/blob/f0eb729bd5f812cb3eec544c6bf980e9b368bf80/core/runtime/impl/tagged_transaction_queue_impl.cpp#L39)

Reply:
Good question. I better change it to the way it is in the Core API, but perhaps if there's a source where it can be checked, it's better to check.